logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/MembersToMentionUseCase.kt

```diff
@@ -0,0 +1,61 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.data.conversation.MemberDetails
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserRepository
+import kotlinx.coroutines.flow.first
+
+class MembersToMentionUseCase internal constructor(
+    private val observeConversationMembers: ObserveConversationMembersUseCase,
+    private val userRepository: UserRepository
+) {
+    suspend operator fun invoke(conversationId: ConversationId, searchQuery: String): List<MemberDetails> {
+        val conversationMembers = observeConversationMembers(conversationId).first()
+
+        // TODO apply normalization techniques that are used for other searches to the name (e.g. ö -> oe)
+
+        val usersToSearch = conversationMembers.filter {
+            it.user.id != userRepository.getSelfUser()?.id
+        }
+        if (searchQuery.isEmpty())
+            return usersToSearch
+
+        if (searchQuery.first().isWhitespace())
+            return listOf()
+
+        val rules: List<(MemberDetails) -> Boolean> = listOf(
```

Comment:
Can we have these rules outside the scope of the function, so if we reuse it we created just once? What do you think?

Reply:
We are using `searchQuery` in it, I don't think that we can extract it.
But isn't better to have it internally as temp value than having it as global declared val ?